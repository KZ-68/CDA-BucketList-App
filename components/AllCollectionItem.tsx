import { FaStar } from "react-icons/fa";
import { MdKeyboardArrowRight } from "react-icons/md";

interface AllCollectionItemProps {
    title: string;
    username : string;
    numberGoals : number;
    isLiked : boolean;
    onLikeToggle: () => void;
  }
  
  const AllCollectionItem = ({ title, username, numberGoals, isLiked, onLikeToggle }: AllCollectionItemProps) => {

    let color = "accentColor";
    let starColor = "darkGrey";
    let starBorder = "accentColor"
    if (isLiked === true) {
        color = "secondColor"
        starColor = "secondColor"
        starBorder = "secondColor"
    } else {
        color = 'accentColor'
        starColor = "darkGrey"
        starBorder = "accentColor"
    }

    return (
      <div className ="flex flex-col gap-2" >
        <div className="bg-darkGrey border border-solid rounded-2xl px-4 py-4 "
            style={{ borderColor: `var(--${color})` }}
        >
            <div className="flex items-center justify-between">
                <p className="uppercase text-xl">{title}</p>
                <p className="opacity-75">{username}</p>
            </div>

            <div className="">
                <p style={{ color: `var(--${color})`}}> {numberGoals}  {numberGoals > 1 ? 'Goals' : 'Goal'}</p>
            </div>

            <div className="flex justify-end text-3xl cursor-pointer">
                <MdKeyboardArrowRight />
            </div>
        </div>

        <div className="flex justify-end">
        <button onClick={onLikeToggle} className="cursor-pointer">
            <FaStar className="text-[25px] cursor-pointer hover:fill-accentColor  transition-all" 
                style={{  
                    color: `var(--${starColor})`,
                    stroke: `var(--${starBorder})`,
                    strokeWidth: '25px',
                }} />
        
        </button>
      </div>
        
      </div>
    );
  };
  
  export default AllCollectionItem;