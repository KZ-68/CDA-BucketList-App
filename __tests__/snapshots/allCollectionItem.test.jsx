import { render } from '@testing-library/react';
import AllCollectionItem from '../../components/AllCollectionItem';

it('Rendering correctly', () => {
  const { asFragment } = render(
        <AllCollectionItem
            title={'Test'}
            userId={'test1238484'}
            username={'testuser'}
            numberGoals={3}
            isLiked={true}
            onLikeToggle={() => handleLike(848485185)}
            numberLikes={5}
            goals={[{label:"Test2", id:"goal84515"}, {label:"Test3", id:"goal484152"}]}
        />
    )
  expect(asFragment()).toMatchSnapshot(); 
});

