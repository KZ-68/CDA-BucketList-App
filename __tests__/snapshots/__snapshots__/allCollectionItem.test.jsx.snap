// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-2"
  >
    <div
      class="bg-darkGrey border border-solid rounded-2xl px-4 pt-4 pb-6"
    >
      <div
        class="flex items-center justify-between"
      >
        <p
          class="uppercase text-xl"
        >
          Test
        </p>
        <a
          href="/collections/user/test1238484"
        >
          <p
            class="opacity-75 hover:opacity-100 transition-all"
          >
            testuser
          </p>
        </a>
      </div>
      <div
        class=""
      >
        <p>
           3 Goals
        </p>
      </div>
      <div
        class="flex justify-end text-3xl cursor-pointer"
      >
        <svg
          class="transition-transform duration-300 "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </div>
    </div>
    <div
      class="flex items-end justify-end gap-2"
    >
      <p
        class="flex items-end leading-none "
      >
        5
      </p>
      <button
        class="cursor-pointer"
      >
        <svg
          class="text-[25px] cursor-pointer hover:fill-accentColor  transition-all"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="stroke: var(--secondColor); stroke-width: 25px;"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
